### PR TITLE
Feature/684 PrematureEmployeeCheck

### PR DESCRIPTION
**Pull request by LeitnerMi**
**Prerequisites: Always create two feature branches (frontend + backend) even if your feature only affects one of them**

**DEV:**
> Development
- [x] Acceptance tests by dev [OpenShift feature environment]
  - [x] Moved to "Ready for review"

> Review
- [x] Pull-request reviewed + approved (including Code Coverage) by reviewer [GitHub]
- [x] Inform dev after review through MEGA-PR (by reviewer) [Chat Space]
- [ ] Version upgrade by dev [GitHub Issue Milestone + maven revision]
- [ ] Merge to main by dev [GitHub]
  - [ ]  Re-add SNAPSHOT to version and push on main [maven revision]
  - [ ]  Make sure all feature branches are deleted [frontend + backend]
  - [ ]  Moved to "Ready for deployment"

> Deployment
- [ ] Integration and Unit tests on main environment passed [Workflow]
- [ ] Deployed to the test environment [GitOps Workflow triggered]
  - [ ] Moved to "Ready for testing"

**PO:**
> Testing
- [ ] Acceptance tests on the test enviroment passed
- [ ] Inspected and approved
  - [ ] Moved to "Ready for release"

### CODE REVIEW

File: src/main/resources/db/changelog-premature.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+databaseChangeLog:
+  - changeSet:
+      id: 1700130995809-5
```
Comment (Ollitod) [inline src/main/resources/db/changelog-premature.yaml:3]: do not use generated number

File: src/test/java/com/gepardec/mega/db/repository/PrematureEmployeeCheckRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
```
Comment (Ollitod) [inline src/test/java/com/gepardec/mega/db/repository/PrematureEmployeeCheckRepositoryTest.java:17]: import static org.assertj.core.api.Assertions.assertThat; (check other places)

File: src/test/java/com/gepardec/mega/db/repository/PrematureEmployeeCheckRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    private static final String EMAIL = "max.muster@gepardec.com";
+    private com.gepardec.mega.db.entity.employee.User user;
+
+    @BeforeEach
+    public void initDependentEntities() {
+        user = initializeUserObject();
+    }
+
+    @Test
+    public void addPrematureEmployeeCheck_RETURN_DB_ID() {
```
Comment (Ollitod) [inline src/test/java/com/gepardec/mega/db/repository/PrematureEmployeeCheckRepositoryTest.java:38]: methodName_StateUnderTest_ExpectedResult (all places)

File: src/test/java/com/gepardec/mega/db/repository/PrematureEmployeeCheckRepositoryTest.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        persistUser();
```
Comment (Ollitod) [inline src/test/java/com/gepardec/mega/db/repository/PrematureEmployeeCheckRepositoryTest.java:39]: declare given, when, then (all places)

File: src/test/java/com/gepardec/mega/service/impl/mapper/PrematureEmployeeCheckMapperTest.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+    @Inject
+    PrematureEmployeeCheckMapper prematureEmployeeCheckMapper;
+
+
+    @Test
+    public void mapToDomain() {
+        PrematureEmployeeCheck prematureEmployeeCheck = prematureEmployeeCheckMapper.mapToDomain(createDBPrematureEmployeeCheck(1L));
+
+        assertAll(
```
Comment (Ollitod) [inline src/test/java/com/gepardec/mega/service/impl/mapper/PrematureEmployeeCheckMapperTest.java:28]: SoftAssertions.assertSoftly(softly -> {softly.assertThat(xxxx); ...}) (check all places)

File: src/test/java/com/gepardec/mega/service/impl/mapper/PrematureEmployeeCheckMapperTest.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+                () -> assertThat(prematureEmployeeCheck.getId()).isEqualTo(1L),
```
Comment (Ollitod) [inline src/test/java/com/gepardec/mega/service/impl/mapper/PrematureEmployeeCheckMapperTest.java:29]: add .as("description")

**Review by Ollitod — CHANGES_REQUESTED** [review]
see comments

File: src/test/java/com/gepardec/mega/service/impl/mapper/PrematureEmployeeCheckMapperTest.java (hunk 5 above)
Comment (LeitnerMi) [reply src/test/java/com/gepardec/mega/service/impl/mapper/PrematureEmployeeCheckMapperTest.java:28]: 71x Usages im Projekt. Soll ich die neuen Refactoren? Soll ich alle Refactoren? Soll ichs lassen, damit nicht 2 Varianten im Code sind? @Ollitod

Comment (LeitnerMi) [reply src/test/java/com/gepardec/mega/service/impl/mapper/PrematureEmployeeCheckMapperTest.java:34]: Es gibt nur eine  einzige Usage von SoftAssertions

**Review by Ollitod — APPROVED** [review]

**Pull request merged**